lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -275,12 +283,16 @@ def grafana_create_dashboard(module, data):
                 result['changed'] = True
             else:
                 body = json.loads(info['body'])
-                raise GrafanaAPIException('Unable to update the dashboard %s : %s' % (uid, body['message']))
+                raise GrafanaAPIException(
+                    'Unable to update the dashboard %s : %s' % (uid, body['message']))
     else:
         # create
-        if 'dashboard' not in payload:
-            payload = {'dashboard': payload}
-        r, info = fetch_url(module, '%s/api/dashboards/db' % data['grafana_url'], data=json.dumps(payload), headers=headers, method='POST')
+        if 'id' in payload["dashboard"]:
+            payload["dashboard"].pop("id")
+        if 'uid' in payload["dashboard"]:
+            payload["dashboard"].pop("uid")
```

Comment:
Unless I am mistaken, 'uid' should not be removed from payload['dashboard'] on create. Per grafana documentation  (http://docs.grafana.org/http_api/dashboard/#identifier-id-vs-unique-identifier-uid) you can specify the uid when creating a dashboard in order to keep your dashboard uids consistent across multiple grafana installs.
So the behavior should actually be:
if uid is provided in Ansible task:
payload["dashboard"]["uid"] = provided_uid
else:
Do nothing - it is up to the user to either remove the 'uid' field in the dashboard JSON or set it to their desired value.

Reply:
your rigth ! when I developped this module, there was no `uid` in Grafana, so i've removed the `uid` same as was doing for the `id` field before when adding the Grafana 5 compatibility.

Reply:
@ematysek @seuf 
Thank you for your review and your comments.
I have updated the module regarding your comments. I also modified the documentation about "uid" parameter.

I tested : 
- Dashboard creation
- Dashboard update
- Dashboard delete

I implemented this behaviour : 
- if uid is provided in Ansible task, use it
- unless, use the uid provided in the JSON file

Reply:
OK, so we can override the `uid` of the json with the uid provided in ansible tasks. Good !
LGTM

Reply:
I think this should be done regardless of create/update.
Scenario being I want to update/replace a dashboard, I provide uid in the ansible task and expect the corresponding dashboard with that uid to be updated but for whatever reason my dashboard JSON has a different uid.

Reply:
In fact, it is already managed here : 
`    else:
        if data.get('uid'):
            uid = data['uid']
        elif 'uid' in payload['dashboard']:
            uid = payload['dashboard']['uid']
        else:
            uid = None
`

I remove what I did regarding uid because it is just a duplicate.